src/GraphQL.SystemTextJson/DocumentWriter.cs

```diff
@@ -16,20 +17,36 @@ public DocumentWriter()
         }
 
         public DocumentWriter(bool indent)
-            : this(new JsonSerializerOptions { WriteIndented = indent, PropertyNamingPolicy = JsonNamingPolicy.CamelCase })
+            : this(GetDefaultSerializerSettings(indent))
         {
         }
 
-        public DocumentWriter(JsonSerializerOptions options)
+        public DocumentWriter(Action<JsonSerializerOptions> configureSerializerOptions)
         {
-            _options = options;
+            _options = GetDefaultSerializerSettings(indent: false);
+            configureSerializerOptions?.Invoke(_options);
 
+            ConfigureOptions();
+        }
+
+        public DocumentWriter(JsonSerializerOptions serializerOptions)
+        {
+            _options = serializerOptions ?? throw new ArgumentNullException(nameof(serializerOptions));
+
+            ConfigureOptions();
+        }
+
+        private void ConfigureOptions()
+        {
             if (!_options.Converters.Any(c => c.CanConvert(typeof(ExecutionResult))))
             {
                 _options.Converters.Add(new ExecutionResultJsonConverter());
             }
         }
 
+        private static JsonSerializerOptions GetDefaultSerializerSettings(bool indent)
```

Comment:
```suggestion
        private static JsonSerializerOptions GetDefaultSerializerOptions(bool indent)
```

Sorry, one more change.

Reply:
Done.